src/relay/op/tensor/transform.cc

```diff
@@ -315,5 +313,72 @@ Example::
 .set_support_level(3)
 .add_type_rel("Reshape", ReshapeRel);
 
+Expr MakeSqueeze(Expr data,
+                 Array<IndexExpr> axis) {
+  auto attrs = make_node<SqueezeAttrs>();
+  attrs->axis = std::move(axis);
+  static const Op& op = Op::Get("squeeze");
+  return CallNode::make(op, {data}, Attrs(attrs), {});
+}
+
+TVM_REGISTER_API("relay.op._make.squeeze")
+.set_body([](const TVMArgs& args, TVMRetValue* rv) {
+    runtime::detail::unpack_call<Expr, 2>(MakeSqueeze, args, rv);
+  });
+
+bool SqueezeRel(const Array<Type>& types,
+                int num_inputs,
+                const Attrs& attrs,
+                const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 2);
+  const auto* data = types[0].as<TensorTypeNode>();
+  if (data == nullptr) {
+    return false;
+  }
+  const auto* param = attrs.as<SqueezeAttrs>();
+  std::vector<IndexExpr> ie;
+  if (param->axis.size() == 0) {
+    for (const auto& e : data->shape) {
+      const int64_t* axis_ptr = as_const_int(e);
+      CHECK(axis_ptr != nullptr);
+      if (*axis_ptr != 1) {
+        ie.push_back(e);
+      }
+    }
+  } else {
+    std::vector<std::pair<IndexExpr, bool> > ieb;
+    for (const auto& e : data->shape) {
+      ieb.push_back(std::pair<IndexExpr, bool>(e, true));
+    }
+    for (const auto& e : param->axis) {
+      const int64_t* axis_ptr = as_const_int(e);
+      CHECK(axis_ptr != nullptr);
+      ieb.at(*axis_ptr).second = false;
+    }
+    for (const auto p : ieb) {
+      if (p.second) {
+        ie.push_back(p.first);
+      } else {
+        const int64_t* axis_ptr = as_const_int(p.first);
+        CHECK(axis_ptr != nullptr);
+        CHECK_EQ(*axis_ptr, 1);
```

Comment:
Here the sanity check should print out more meaningful message to help users better understand the error.